mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/kafka/Producer.java

```diff
@@ -9,11 +10,14 @@
 
     private static final Logger logger = LoggerFactory.getLogger(Producer.class);
 
+    private KafkaTemplate<String, Mention> kafkaTemplate;
+
     @Autowired
-    private KafkaTemplate<String, String> kafkaTemplate;
+    public Producer(KafkaTemplate<String, Mention> kafkaTemplate) {
+    }
 
-    public void send(String message) {
-        logger.info("sending message='{}", message);
-        kafkaTemplate.send("mentions", "Hidwdae", message);
+    public void send(Mention mention){
+        logger.info("sending message='{}", mention);
+        kafkaTemplate.send("mentions",String.valueOf(System.currentTimeMillis()), mention);
```

Comment:
Why did you choose `String.valueOf(System.currentTimeMillis())` specifically?

Reply:
Just needed a unique key.

Reply:
A unique key could be queryId+resourceId, for example. If you want time, maybe just take the time of the mention?

Reply:
I don't know if the time of the Mention is unique, because it is coming from twitter. But queryID+resourceID sounds good.

Reply:
But I don't have a resource there. What is wrong with `String.valueOf(System.currentTimeMillis()`?

Reply:
There is no requirement for a unique key, is there? You should choose the key base on what you want to partition by. For example, if you had a technical requirement to process mentions in the order they were created by query id, you would choose the query id as the key. That way, all mentions from the same query will go to the same partition. Kafka guarantees the order per partition, but not per topic.

But I don't think you have any ordering requirement, so just choose what you want.

Reply:
If you want to define mentionId=queryId-resourceId then you would have to add it to the mention object when you create it (there you should have both the query and the resource).

If you go for time, I don't see a problem with the mention time. If you don't want that, you should probably favour something like `Instant.now()` over the system time thing.

Reply:
But the QueryID isn't unique

Reply:
Didn't saw that last comment before writing the comment.

Reply:
Problem with mention time is, that I don't know if this time is unique. The mention `date` is set by an attribute `createdAt`, that I receive from twitter.

Reply:
I can use `Instant.now()`, but if I'm right it just uses `System.currentTimeMillis()` under the hood.

![image](https://user-images.githubusercontent.com/26352108/64178085-14586e00-ce61-11e9-90d4-489df401d774.png)

Reply:
I changed it to `Instant.now()`

Reply:
It's good style to use the time api instead of interacting with `System` directly unless you have a good reason.

>Problem with mention time is, that I don't know if this time is unique

As I said, I don't think that's a requirement.

Reply:
Yes, I've overread that the first time. Whats the purpose of the key then, when it doesn't have to be unique?